### PR TITLE
Unreleased changes in changelog

### PR DESCRIPTION
**Pull request by prckent**
Updating the changelog with significant changes to the development version since the last release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,14 +2,47 @@
...
+  performance.  
+* Improved stability of GPU matrix inversion for large runs.
+* Ongoing improvements to output to improve readability.
+* Initial interface to PySCF for real space QMC trial wavefunctions.
+* Enabled use of HDF5 files for Gaussian based wavefunctions
+  with SoA implementation.
+* Added Appendix to manual listing all known QMCPACK publications.
+  This will be updated on an ongoing basis. Please advise of any
+  missing publications.
+* Improved Momentum distribution estimator: optimized and
```
Comment (markdewing) [inline CHANGELOG.md]: and ...?

Comment (prckent) [reply CHANGELOG.md]: Fixed. Was thinking about the bugfix, but that was in 3.3.0.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,14 +2,47 @@
...
 
+## [Unreleased]
+
+* Size consistent t-moves implemented (Casula 2010 algorithm).
+  Enabled via nonlocalmoves parameter, see manual.
+* Bugfix: For non-git builds, build process failed on some systems
+  due to git-rev.h handling.
+* Optimized load balancing in DMC. Command line option async_swap
+  removed. Parameter use_nonblocking now disables non-blocking MPI
+  load balancing.
```
Comment (ye-luo) [inline CHANGELOG.md]: Parameter use_nonblocking now enables (by default) non-blocking MPI load balancing.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,14 +2,47 @@
...
+  performance.  
+* Improved stability of GPU matrix inversion for large runs.
+* Ongoing improvements to output to improve readability.
+* Initial interface to PySCF for real space QMC trial wavefunctions.
+* Enabled use of HDF5 files for Gaussian based wavefunctions
+  with SoA implementation.
+* Added Appendix to manual listing all known QMCPACK publications.
+  This will be updated on an ongoing basis. Please advise of any
+  missing publications.
+* Optimized momentum distribution estimator.
```
Comment (ye-luo) [inline CHANGELOG.md]: Supported by SoA and 1/2/3 body Jastrows.

Comment (prckent) [reply CHANGELOG.md]: Updated

File: CHANGELOG.md (hunk 2 above)
Comment (prckent) [reply CHANGELOG.md]: Fixed

**Pull request merged**